datafusion-federation/src/schema_cast/record_convert.rs

```diff
@@ -153,4 +175,43 @@ mod test {
         let result = try_cast_to(batch_input(), to_schema()).expect("converted");
         assert_eq!(3, result.num_rows());
     }
+
+    fn large_string_schema() -> SchemaRef {
+        Arc::new(Schema::new(vec![
+            Field::new("a", DataType::Int32, false),
+            Field::new("b", DataType::LargeUtf8, false),
+            Field::new("c", DataType::LargeUtf8, false),
+        ]))
+    }
+
+    fn large_string_to_schema() -> SchemaRef {
+        Arc::new(Schema::new(vec![
+            Field::new("a", DataType::Int64, false),
+            Field::new("b", DataType::LargeUtf8, false),
+            Field::new("c", DataType::Timestamp(TimeUnit::Microsecond, None), false),
+        ]))
+    }
+
+    fn large_string_batch_input() -> RecordBatch {
+        RecordBatch::try_new(
+            large_string_schema(),
+            vec![
+                Arc::new(Int32Array::from(vec![1, 2, 3])),
+                Arc::new(LargeStringArray::from(vec!["foo", "bar", "baz"])),
+                Arc::new(LargeStringArray::from(vec![
+                    "2024-01-13 03:18:09.000000",
+                    "2024-01-13 03:18:09",
+                    "2024-01-13 03:18:09.000",
+                ])),
+            ],
+        )
+        .expect("record batch should not panic")
+    }
+
+    #[test]
+    fn test_large_string_to_timestamp_conversion() {
+        let result =
+            try_cast_to(large_string_batch_input(), large_string_to_schema()).expect("converted");
+        assert_eq!(3, result.num_rows());
```

Comment:
Can you also assert the results? Using a macro like `assert_batches_eq` from the main datafusion project can make it easy.